Platform/OpenCanopy/Views/BootPicker.c

```diff
@@ -718,42 +726,180 @@ GLOBAL_REMOVE_IF_UNREFERENCED GUI_OBJ mBootPickerView = {
   INITIALIZE_LIST_HEAD_VARIABLE (mBootPicker.Hdr.Link)
 };
 
+STATIC
+UINT8
+AppleDiskLabelImagePalette[256] = {
+  [0x00] = 255,
+  [0xf6] = 238,
+  [0xf7] = 221,
+  [0x2a] = 204,
+  [0xf8] = 187,
+  [0xf9] = 170,
+  [0x55] = 153,
+  [0xfa] = 136,
+  [0xfb] = 119,
+  [0x80] = 102,
+  [0xfc] = 85,
+  [0xfd] = 68,
+  [0xab] = 51,
+  [0xfe] = 34,
+  [0xff] = 17,
+  [0xd6] = 0
+};
+
+EFI_STATUS
+DecodeAppleDiskLabelImage (
+  OUT GUI_IMAGE *Image,
+  IN  UINT8     *RawData,
+  IN  UINT32    DataLength
+  )
+{
+  UINT32 PixelIdx;
+
+  if (RawData == NULL || DataLength < 5) {
+    return EFI_INVALID_PARAMETER;
+  }
```

Comment:
We generally use ASSERT() to ensure function preconditions. DataLength < 5 may be a precondition, but then I'd say to remove the condition from the file reading function to have one single spot of validation. Also, magic constant.